pkg/policy/rule.go

```diff
@@ -58,46 +58,44 @@ func (r *rule) String() string {
 	return fmt.Sprintf("%v", r.EndpointSelector)
 }
 
-func (l4 *L4Filter) mergeCachedSelectors(from *L4Filter, selectorCache *SelectorCache) {
-	for _, cs := range from.CachedSelectors {
-		if l4.CachedSelectors.Insert(cs) {
-			// Update selector owner to the existingFilter
-			selectorCache.ChangeUser(cs, from, l4)
+func (epd *PerSelectorPolicy) appendL7WildcardRules(ctx *SearchContext) *PerSelectorPolicy {
+	switch {
```

Comment:
nits: what about the default case, including go extensions?

Reply:
Looking at the other L7 handling piece in this file, I guess we don't support merging for such L7 rules. That said I think it's probably worth raising at least a policy trace line, but probably also a warning of some kind that you're doing something not-supported in that case?

Reply:
go extensions do not need a L7 rule to be triggered as the parser type/id is explicitly invoked in the policy. Even if no rules acts as a wildcard, an empty set would need to be appended here. Will fix!

Reply:
Added a commit to address this :-)